HACKING.md

```diff
@@ -1,114 +1,233 @@
-go modules feature. Use:
 
-    export GO111MODULE=off
+### Getting the snapd sources
 
-for this.
+The easiest way to get the source for `snapd` is to clone the GitHub repository
+in a directory where you have read-write permissions, such as your home
+directory.
 
+    cd ~/
+    git clone https://github.com/snapcore/snapd.git
+    cd snapd
 
-### Getting the snapd sources
+This will allow you to build and test `snapd`. If you wish to contribute to
+the `snapd` project, please see the Contributing section.
+
+### Contributing
+
+Contributions are always welcome!
+
+Please make sure that you sign the Canonical contributor agreement [here](
+http://www.ubuntu.com/legal/contributors).
+
+Complete requirements can be found in CONTRIBUTING.md.
+
+Contributions are submitted through a Pull Request created from a fork of the
+`snapd` repository (under your GitHub account). 
+
+Start by creating a fork of the `snapd` repository on GitHub.
 
-The easiest way to get the source for `snapd` is to use the `go get` command.
+Add your fork as an additional remote to an already cloned `snapd` main
+repository. Replace `<user>` with your GitHub account username.
 
-    go get -d -v github.com/snapcore/snapd/...
+```
+cd ~/snapd
+git remote add fork git@github.com:<user>/snapd.git
+```
 
-This command will checkout the source of `snapd` and inspect it for any unmet
-Go package dependencies, downloading those as well. `go get` will also build
-and install `snapd` and its dependencies. To also build and install `snapd`
-itself into `$GOPATH/bin`, omit the `-d` flag. More details on the `go get`
-flags are available using
+Create a working branch on which to commit your work. Replace
+`<branchname>` with a suitable branch name.
 
-    go help get
+```
+git checkout -b <branchname>
+```
 
-At this point you will have the git local repository of the `snapd` source at
-`$GOPATH/src/github.com/snapcore/snapd`. The source for any
-dependent packages will also be available inside `$GOPATH`.
+Make changes to the repository and commit the changes to your
+working branch. Push the changes to your forked `snapd` repository.
 
-### Dependencies handling
+```
+git commit -a -m "commit message"
+git push fork <branchname>
+```
 
-Go dependencies are handled via `go mod`.
+Create the Pull Request for your branch on GitHub.
 
-Other dependencies are handled via distribution packages and you should ensure
-that dependencies for your distribution are installed. For example, on Ubuntu,
-run:
+This complete process is outlined in the GitHub documentation [here](
+https://docs.github.com/en/github/collaborating-with-pull-requests).
 
-    sudo apt-get build-dep ./
+We value good tests, so when you fix a bug or add a new feature we highly
+encourage you to add tests.
 
-### Building
+For more information on testing, please see the Testing section.
 
-To build, once the sources are available and `GOPATH` is set, you can just run
+### Build host dependencies
```

Comment:
This might be because I'm not a native speaker, but to me this sentence means that this section will explain how to build my host dependencies. Maybe "Installing build dependencies" is clearer?